From 22cc50481d58bc50d21a577035bbb949dd89a34e Mon Sep 17 00:00:00 2001
From: CI Assistant <yossi.ilkanaev@intel.com>
Date: Wed, 8 Aug 2018 12:22:57 +0200
Subject: [PATCH] Merge pull request #117 in SW_UGW/linux from
 feature/DRVLIB_SW-845-ethernet-driver-add-multiqueue-support to master

* commit 'c86721e82cf59ef0b68b8e207804de6e22eec2f9':
  DRVLIB_SW-845: ethernet driver add multiqueue support
---
 drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c | 5 ++++-
 drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.h | 1 +
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
index 2792e2a79697..07f3fc17d116 100644
--- a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
+++ b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
@@ -1210,7 +1210,8 @@ static int xrx500_of_iface(struct xrx500_hw *hw, struct device_node *iface,
 	int ret;
 
 	/* alloc the network device */
-	hw->devs[hw->num_devs] = alloc_etherdev(sizeof(struct ltq_eth_priv));
+	hw->devs[hw->num_devs] = alloc_etherdev_mq(sizeof(struct ltq_eth_priv),
+						   g_soc_data.queue_num);
 
 	if (!hw->devs[hw->num_devs]) {
 		pr_debug("allocated failed for interface %d\n",
@@ -2123,11 +2124,13 @@ static int ltq_eth_drv_remove(struct platform_device *pdev)
 static const struct ltq_net_soc_data xrx500_net_data = {
 	.need_defer = true,
 	.hw_checksum = true,
+	.queue_num = 1,
 };
 
 static const struct ltq_net_soc_data falconmx_net_data = {
 	.need_defer = false,
 	.hw_checksum = false,
+	.queue_num = 8,
 };
 
 static const struct of_device_id ltq_eth_drv_match[] = {
diff --git a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.h b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.h
index 27b166dc136f..87101d873df8 100644
--- a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.h
+++ b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.h
@@ -121,5 +121,6 @@ extern int g_xway_gphy_fw_loaded;
 struct ltq_net_soc_data {
 	bool need_defer;
 	bool hw_checksum;
+	unsigned int queue_num;
 };
 #endif /* _LANTIQ_ETH_DRV_H_ */
